rfd/0136-modern-signature-algorithms.md

```diff
@@ -0,0 +1,876 @@
+held in GCP KMS.
+In the future we will likely support more KMS services.
+
+Teleport currently has these CAs:
+
+#### User CA
+
+keys: ssh, tls
+
+uses: user ssh cert signing, user tls cert signing, ssh hosts trust this CA
+
+* current/`legacy` SSH key type: 2048-bit RSA
+* proposed `balanced-v1` key type: Ed25519
+* proposed `fips-v1` key type: ECDSA with NIST P-256
+* proposed `hsm-v1` key type: ECDSA with NIST P-256
+* reasoning:
+  * Ed25519 is currently considered by multiple sources to be the best
+    algorithm for SSH
+  * ECDSA with P-256 has Go BoringCrypto support, Ed25519 doesn't
+  * Ed25519 support was added to OpenSSH 6.5 in January 2014, *before* SHA-2
+    hash support was added for RSA keys in OpenSSH 7.2
+  * Some SSH clients other than OpenSSH have very limited support for newer
+    protocols - we will need to make sure this is very visbible as a possibly
+    breaking change.
+
+* current/`legacy` TLS key type: 2048-bit RSA
+* proposed `balanced-v1` key type: ECDSA with NIST P-256
+* proposed `fips-v1` key type: ECDSA with NIST P-256
+* proposed `hsm-v1` key type: ECDSA with NIST P-256
+* reasoning:
+  * external tools may need to load user X.509 certificates signed by this CA,
+    e.g. for application access, and Ed25519 support is generally spotty
+
+#### Host CA
+
+keys: ssh, tls
+
+uses: host ssh cert signing, host tls cert signing, ssh clients trust this CA
+
+* current/`legacy` SSH key type: 2048-bit RSA
+* proposed `balanced-v1` key type: Ed25519
+* proposed `fips-v1` key type: ECDSA with NIST P-256
+* proposed `hsm-v1` key type: ECDSA with NIST P-256
+* reasoning:
+  * Ed25519 is currently considered by multiple sources to be the best
+    algorithm for SSH
+  * Ed25519 support was added to OpenSSH 6.5 in January 2014, *before* SHA-2
+    hash support was added for RSA keys in OpenSSH 7.2
+  * ECDSA with P-256 has Go BoringCrypto support, Ed25519 doesn't
+  * ECDSA is supported by HSMs, Ed25519 isn't
+
+* current/`legacy` TLS key type: 2048-bit RSA
+* proposed `balanced-v1` key type: ECDSA with NIST P-256
+* proposed `fips-v1` key type: ECDSA with NIST P-256
+* proposed `hsm-v1` key type: ECDSA with NIST P-256
+* reasoning:
+  * it's possible we may want an external tool to trust host certs, and support
+    for Ed25519 X.509 certs is generally spotty
+
+#### Database CA
+
+keys: tls
+
+uses:
+
+* signs (often) long-lived db cert used to authenticate db to database service
+* signs short-lived proxy cert used to authenticate proxy to database service
+* signs snowflake JWTs
+* self-hosted databases (and Snowflake) trust this CA
+
+* current/`legacy` TLS key type: 2048-bit RSA
+* proposed `balanced-v1` key type: 2048-bit RSA
+* proposed `fips-v1` key type: 2048-bit RSA
+* proposed `hsm-v1` key type: 2048-bit RSA
+* reasoning:
+  * some database protocols still require RSA, reduce friction by keeping it as
+    the default
+
+#### OpenSSH Host CA
```

Comment:
Nit: maybe name this section `OpenSSH CA`? It gives the impression we sign SSH host keys for agentless nodes with a different CA than the Host CA, which isn't the case.

Reply:
Yeah good catch thanks, I originally misunderstood what the OpenSSH CA was doing and forget to update the heading when I figured it out.